lib/tests/streamlit/elements/exception_test.py

```diff
@@ -150,24 +160,94 @@ def func_with_error():
             f"Stack does not have length {original_stack_len}: {proto.stack_trace}",
         )
 
-    def test_uncaught_app_exception(self):
-        err = None
-        try:
-            st.format("http://not_an_image.png", width=-1)
-        except Exception as e:
-            err = UncaughtAppException(e)
-        self.assertIsNotNone(err)
-
-        # Marshall it.
-        proto = ExceptionProto()
-        exception.marshall(proto, err)
-
-        for line in proto.stack_trace:
-            # assert message that could contain secret information in the stack trace
-            assert "module 'streamlit' has no attribute 'format'" not in line
-
-        assert proto.message == _GENERIC_UNCAUGHT_EXCEPTION_TEXT
-        assert proto.type == "AttributeError"
+    def test_uncaught_app_exception_default_setting_community_cloud(self):
+        """
+        In community cloud, we show a generic error message, the trace and the type.
+        Community cloud uses False. This is the same results as "stacktrace".
+        """
+        with testutil.patch_config_options({"client.showErrorDetails": False}):
+            err = None
+            try:
+                st.format("http://not_an_image.png", width=-1)
+            except Exception as e:
+                err = e
+            self.assertIsNotNone(err)
+
+            # Marshall it.
+            proto = ExceptionProto()
+            exception.marshall(proto, err, is_uncaught_app_exception=True)
+
+            assert proto.message == _GENERIC_UNCAUGHT_EXCEPTION_TEXT
+            assert len(proto.stack_trace)
+            assert proto.type == "AttributeError"
+
+    def test_uncaught_app_exception_full_config_setting(self):
+        with testutil.patch_config_options({"client.showErrorDetails": "full"}):
+            err = None
+            try:
+                st.format("http://not_an_image.png", width=-1)
+            except Exception as e:
+                err = e
+            self.assertIsNotNone(err)
+
+            # Marshall it.
+            proto = ExceptionProto()
+            exception.marshall(proto, err, is_uncaught_app_exception=True)
+
+            assert proto.message == "module 'streamlit' has no attribute 'format'"
+            assert len(proto.stack_trace)
```

Comment:
nit: do you mean `assert len(proto.stack_trace) > 0`? If yes, I thing writing it out explicitly helps to understand the intention